Cython/Utility/ModuleSetupCode.c

```diff
@@ -1588,15 +1588,15 @@ static int __Pyx_check_binary_version(void);
 /////////////// CheckBinaryVersion ///////////////
 
 static int __Pyx_check_binary_version(void) {
-    char ctversion[4], rtversion[4];
-    PyOS_snprintf(ctversion, 4, "%d.%d", PY_MAJOR_VERSION, PY_MINOR_VERSION);
-    PyOS_snprintf(rtversion, 4, "%s", Py_GetVersion());
-    if (ctversion[0] != rtversion[0] || ctversion[2] != rtversion[2]) {
+    int ct_major_version = PY_MAJOR_VERSION, ct_minor_version = PY_MINOR_VERSION;
+    int rt_major_version, rt_minor_version;
+    sscanf(Py_GetVersion(), "%d.%d", &rt_major_version, &rt_minor_version);
+    if (ct_major_version != rt_major_version || ct_minor_version != rt_minor_version) {
```

Comment:
I think the older code is better since it does not try to parse. Shouldn't this work?
```suggestion
    char ct_version[5], rt_version[5];
    PyOS_snprintf(ctversion, 5, "%d.%d", PY_MAJOR_VERSION, PY_MINOR_VERSION);
    PyOS_snprintf(rtversion, 5, "%s", Py_GetVersion());
    if (strncmp(ctversion, rtversion, 4) != 0) {
```

Reply:
.. and then the error message change below should be reverted

Reply:
I agree with the general principle (just compare the strings). `Py_GetVersion()` will have spaces and possibly letters immediately after the version number so I think it needed a bit of selective copying to just pick out the version number though (so I haven't used quite this code)